### ISSUE TITLE
Low risk (L2): lack of check of actual distributor in `manualDeposit(address,address,uint256)`

### ISSUE THREAD

**Issue by petrovska-petro** [BalancerMaxis/ChildGaugeInjectorV2#23]
**Severity:** low risk. Ultimately, will leave to revert.

**Context:** in `manualDeposit` method the caller could point to any gauge given the flexibility of the parameters, but not necessarily being a distributor for that gauge and `reward_token`. It will lead to reverts.

**Recommendation:** add a conditional check or handle gracefully when the following condition is not true

```
gaugeContract.reward_data(reward_token).distributor == address(this)
```

Also, we acknowledge that lack of balance of `reward_token` may be reverted via `revert SafeERC20FailedOperation`, but internally we consider that revert method to be slightly abstract, recommendation to consider here will be to check that actual enough `amount` is present in the contract at the time of calling.

## Review stage 

Balancer Maxis:
- [ ]  Acknowledge
- [ ]  Fixed

Onchainification Labs:
- [ ]  Acknowledge correction/mitigation

**Comment by Tritium-VLK** [comment]
This is about basically catching an exception and throwing another revert in an admin operation that has never been used in over a year of operation of Injector V1.  

This is a simple helper contract, while catching/reexpressing all the errors may make it cleaner code, it also adds LoC and some complexity.

Good feedback, but I think for this audit/review we're looking to make sure the code is SaFu and has good UX for normal operations.  This is a bit outside the scope of what I am working towards at the moment and in some ways opens up a can of worms.

**Comment by petrovska-petro** [comment]
@Tritium-VLK yes, handling on a try/catch approach could work, and indeed it will increase LoC. But we understand that it is not something important at this stage to be required

> [!IMPORTANT] 
> something regarding this same method that went unnoticed initially is that if for example a manual deposit is done into a gauge which is within the active gauge list of the injector it will not increase the `periodNumber` or its accounting of the balances, while `InjectTokenAddress` = `reward_token`

Then, we advise to reconsider if you would like to manually deposit **only** to gauges which are not into the active list in storage or only for different tokens than `InjectTokenAddress`, or in the other hand, this dynamic it is something you are comfortable with it, which may lead to break the current `periodNumber` accounting and as well the token balance in the contract to cover the whole schedules

Advise in your considerations on the notes above

**Comment by Tritium-VLK** [comment]
a manual injection is to modify the running schedule somehow.  It changes the flow of rewards anyway.  The point of inject funds is to be able to potentally shift the date of a running injection or something by injecting into an already running schedule.

These are deep details of incentives management specific to curve gauges.  The expectation is that only someone who is an expert in this would use these functions.  The normal user interface is just setting schedules/managing, having them pay out, and maybe sweeping some dust.  The contract should help the admin understand total spend vs injector funding.

**Comment by petrovska-petro** [comment]
Understood, then we can acknowledge that by design not updating the `periodNumber` is safe and only experts will manage this admin method and aware of its implications into the other storage variables.

Marking as "Acknowledge correction/mitigation", closing the issue.
